osquery/tables/system/windows/users.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma comment(lib, "netapi32.lib")
+
+#include <iostream>
+
+#define _WIN32_DCOM
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+// clang-format off
+#include <LM.h>
+// clang-format on
+#include <Shlobj.h>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+
+#include "osquery/tables/system/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+void genUsers(QueryData& results) {
+  NET_API_STATUS nEnumStatus;
+  LPUSER_INFO_3 pUserBuffer = nullptr;
```

Comment:
A few of these can have their scope reduced. Maybe the content within the `do`, `while` can be placed into another method that returns a `Status`?